packages/ckeditor5-image/src/imagecaption/toggleimagecaptioncommand.js

```diff
@@ -0,0 +1,168 @@
+/**
+ * @license Copyright (c) 2003-2021, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module image/imagecaption/toggleimagecaptioncommand
+ */
+
+import { Command } from 'ckeditor5/src/core';
+import { Element } from 'ckeditor5/src/engine';
+
+import ImageBlockEditing from '../image/imageblockediting';
+import { isImage, isImageInline } from '../image/utils';
+import { getCaptionFromImageModelElement, getCaptionFromModelSelection } from './utils';
+
+/**
+ * The toggle image caption command.
+ *
+ * This command is registered by {@link module:image/imagecaption/imagecaptionediting~ImageCaptionEditing} as the
+ * `'toggleImageCaption'` editor command.
+ *
+ * Executing this command:
+ *
+ * * either adds or removes the image caption of a selected image (depending on whether the caption is present or not),
+ * * removes the image caption if the selection is anchored in one.
+ *
+ *		// Toggle the presence of the caption.
+ *		editor.execute( 'toggleImageCaption' );
+ *
+ * **Note**: Upon executing this command, the selection will be set on the image if previously anchored in the caption element.
+ *
+ * **Note**: You can move the selection to the caption right away as it shows up upon executing this command by using
+ * the `focusCaptionOnShow` option:
+ *
+ *		editor.execute( 'toggleImageCaption', { focusCaptionOnShow: true } );
+ *
+ * @extends module:core/command~Command
+ */
+export default class ToggleImageCaptionCommand extends Command {
+	/**
+	 * @inheritDoc
+	 */
+	refresh() {
+		const editor = this.editor;
+
+		// Only block images can get captions.
+		if ( !editor.plugins.has( ImageBlockEditing ) ) {
+			this.isEnabled = this.value = false;
+
+			return;
+		}
+
+		const selection = editor.model.document.selection;
+		const selectedElement = selection.getSelectedElement();
+
+		if ( !selectedElement ) {
+			const ancestorCaptionElement = getCaptionFromModelSelection( selection );
+
+			this.isEnabled = this.value = !!ancestorCaptionElement;
+
+			return;
+		}
+
+		// Only block images have captions.
+		this.isEnabled = isImage( selectedElement ) || isImageInline( selectedElement );
```

Comment:
The command should be enabled on both types of images to be able to toggle image types.